Integrate new header design for product page

jolelievre opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Integrate new header design based on the most recent mockup and add a few new features:<br>- display prices (with tax rate if specified)<br>- display stock (with color code depending on the stock level)<br>- display product references<br>- new component to switch product type, looks like the create product modal
| Type?             | new feature
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #15330
| How to test?      | Check that the value in the header are updated as expected after submit (no automatic js update is handled) You can also test the component to switch product type
| Possible impacts? | ~


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27233)
<!-- Reviewable:end -->

### New display preview
![Capture d’écran 2022-01-06 à 12 47 05](https://user-images.githubusercontent.com/13801017/148409124-9707b3f4-4a51-42a8-a03e-59301ef495be.png)

### Specs
https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type

### Breaking Changes

- `LocalizationExtension` now has three more arguments in its constructor (built by DI)
- `ProductFormDataProvider` has many less arguments in its constructor since creation data has been cleaned a lot
- `ProductTypeUpdater` has a new parameter in its constructor `ProductStockUpdater $productStockUpdater`
- `ProductStockUpdater` has a new parameter `MovementReasonRepository $movementReasonRepository`
- `GetProductForEditingHandler` has a new parameter `Configuration` adapter service
- `PricingType` form type has a new parameter `$taxEnabled`

jolelievre committed:
Create new product header display, and also adapt tabs to be fullwidth

jolelievre committed:
Extract toolbar from product header, and separate elements in the header

jolelievre committed:
Use cart image format in product cover to have a bigger image

jolelievre committed:
Create ProductTypeType form type which is used for creation form, and product type selection modal

jolelievre committed:
Close callback returns void

jolelievre committed:
Product type selector component uses a configurable selector to be used in multiple cases

jolelievre committed:
Convert ProductTypeManager component into TypeScript ProductTypeSwitcher, the selection now happens in a modal which display the ProductTypeSelector component

jolelievre committed:
Reorganize footer and refacto display, only active and save are displayed on the right now

jolelievre committed:
No more labels in toolbar buttons

jolelievre committed:
Add price_format twig filter to render price with CLDR

jolelievre committed:
Integrate product summary in the header, display price and stock

jolelievre committed:
Handle dynamic stock display

jolelievre committed:
Display tax rate with price tax included

jolelievre committed:
Integrate references design

jolelievre committed:
Make LayoutExtension more resilient to incorrect context values

NeOMakinG commented:
Seems like it breaks the CI, while front-end modifications looks ok to me

jolelievre committed:
Update expected resulsts for cover image

matks commented on src/PrestaShopBundle/Form/Admin/Sell/Product/ProductTypeType.php:
```diff
@@ -0,0 +1,66 @@
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ProductTypeType extends ChoiceType
```
ProductTypeType ? 😄 
is that intended and if yes, could we find a name less confusing ?

jolelievre replied:
I couldn't find better, all form types class names end with Type by convention
And this one is used to edit the "product type" So even though I agree it's not ideal it's the less worse choice 😅

Calling it simply `ProductType` would be confusing because based on the usual convention it would imply a form type for Product

matks replied:
Agreed.

Only idea I have is to use something else for "product type". "Product classification"? "Product specificity"?

matks commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Blocks/header.html.twig:
```diff
@@ -22,59 +22,87 @@
+            {% for group_label, choice in productForm.pricing.tax_rules_group_id.vars.choices %}
+              {% if choice is selectedchoice(productForm.pricing.tax_rules_group_id.vars.value) %}
+                {% set taxRate = choice.attr['data-tax-rate'] %}
+                {% dump(choice) %}
```
Looks like you forgot some `dump()` 😄

jolelievre replied:
Nice catch 😅

Hlavtox commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,112 @@ $product-page-padding-bottom: 80px !default;
+        width: $product-cover-size;
+        height: $product-cover-size;
+
+        img {
```
I would just use .img-fluid on the image, no need for these styles. Set width and height parameters on the image directly to improve CLS in modern browsers.

jolelievre replied:
Ok I removed that part, because you're right my image already used the `img-fluid` class However I still need to limit the size of its container But it's the only one that is limited now

Hlavtox replied:
Yes, play with the width of the container, let everything inside adapt. :-)

Hlavtox commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,112 @@ $product-page-padding-bottom: 80px !default;
+
+      .image_preview-widget {
+        width: $product-cover-size;
+        height: $product-cover-size;
```
Avoid using fixed height. If image is bigger for any reason, you will have trouble, just let it flow ;-)

jolelievre replied:
The widget container has a fixed height, but the image inside is floating It has auto-size but max-width and height The purpose is to let it adapt inside a restricted container

Hlavtox commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Combination/external_tab.html.twig:
```diff
@@ -24,7 +24,7 @@
  *#}
 
 <div id="external-combination-tab" class="form-horizontal row justify-content-md-center d-none">
```
You can remove `row justify-content-md-center` and the inside `<div class="col">`

jolelievre replied:
Nice catch, thanks!

jolelievre replied:
I think you are pleased to see all the useless divs I removed in this PR 😅

Hlavtox replied:
Yes! 😄 You know I am watching 👀

jolelievre committed:
Update behat test for ne type of cover

jolelievre committed:
Reorganize product type selectors

jolelievre committed:
Clean external tab layout

jolelievre committed:
Simplify header cover css and remove margin under header form

jolelievre committed:
Clean debug dumps

Hlavtox commented:
Joe don't forget we need to somehow integrate this. 🤔 Or just merge this and we can play with the footer later. 👍

![desktop](https://user-images.githubusercontent.com/6097524/148379555-d895a66a-357e-46a4-912b-c711c2c8c222.JPG)
![mobile](https://user-images.githubusercontent.com/6097524/148379560-e7291a7e-2a96-4e7a-a103-fd81ab8d5385.JPG)

Hlavtox commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php:
```diff
@@ -368,7 +367,6 @@ private function renderEditProductForm(FormInterface $productForm, int $productI
             'productForm' => $productForm->createView(),
             'statsLink' => $this->getAdminLink('AdminStats', ['module' => 'statsproduct', 'id_product' => $productId]),
```
@jolelievre Can you check this line? I am on develop and the button is displayed even if I uninstalled the related stat modules. Maybe we could check if the module is active. ;-)

jolelievre replied:
I added a check, the link is only added when module `statsproduct` is installed and enabled

jolelievre committed:
Handle radius on product tabs, and external tab as well Homogenize the style in external tab to match others

jolelievre committed:
Only display stats link when module statsproduct is installed and enabled

jolelievre committed:
Extract some of the css rules outside of the v2 special class

Hlavtox commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -1275,50 +1271,58 @@ $product-page-padding-bottom: 80px !default;
-        margin: 0.25rem 0;
-        text-align: left;
+  select {
+    display: none;
```
❗

NeOMakinG commented on admin-dev/themes/new-theme/js/pages/product/product-const.ts:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+export default {
+  PRODUCT_TYPE: {
+    STANDARD: 'standard',
+    PACK: 'pack',
+    VIRTUAL: 'virtual',
+    COMBINATIONS: 'combinations',
+  },
+};
```
product-const.ts looks like it's not a good name, it's redundant because it's already inside the product folder, I would call it `constants` and that's it

jolelievre replied:
Ok but then it means we should also rename `product-event-map.js` `product-map.js` and so on
I'm fine with it but maybe in a future PR I just renamed the constant file for now

NeOMakinG replied:
Well, you're damn right about this, but then another question happens about components name 😆

NeOMakinG commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,105 @@ $product-page-padding-bottom: 80px !default;
   }
 
   // Components
+  .product-header-v2 {
```
Do we need to version the header inside a classname? can't we just use product-header?

NeOMakinG replied:
Seen the comment after this line, idc!

jolelievre replied:
Hmm that's weird I had noted some comments for this but they seem to have disappeared
Ah no it's on the `.product-page-v2` class but it's the same principle for this one

For now we have two different pages at the same time but they share the same CSS so sometimes I need to target only the v2 specifically to avoid breaking the legacy one Ideally we should have created a dedicated new scss file but I guess we didn't because it was more convenient this way

Anyway this will be cleaned later when the legacy page is remove, I won't need this extra scope for v2 page

NeOMakinG commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,105 @@ $product-page-padding-bottom: 80px !default;
+  .product-header-v2 {
+    display: flex;
+    flex-direction: row;
+    gap: 25px;
```
```suggestion
    gap: 1.5625rem;
```

I think if we use `rem` on margin and paddings, no reason to not use it on gaps, as we wants it to adapt on font-size change

jolelievre replied:
You're right I copied this from the mockup directly I converted the other ones but I forgot this one, my bad

NeOMakinG commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,105 @@ $product-page-padding-bottom: 80px !default;
+    $product-cover-size: 110px;
+
+    .product-header-cover {
+      width: $product-cover-size;
+      max-width: $product-cover-size;
+      height: $product-cover-size;
+      max-height: $product-cover-size;
```
Max width and max-height are useless, the width and height can't change if they are fixed

Why no just use img-fluid? The image should adapt depending on his width but maybe there are some PHP stuff that force the size anyway

jolelievre replied:
Actually, since I'm in a flex container the max-width and height are the most important Without it the div was too large despite the presence of width/height

jolelievre replied:
As I already explained to @Hlavtox the image is already in `img-fluid` This class doesn't target the image it targets its div container which must be restricted and keep a fixed size (the flex container tries to make it larger if I don't restrict it)

Hlavtox replied:
`flex-grow:0;`

jolelievre replied:
Thanks @Hlavtox `flex-grow: 0;` works nicely indeed!

I just left a max-width though because the image I use is still wider than the expected cover. We don't have an image type with 110px size and I don't think it's worth creating a new image type for this.

jolelievre committed:
Address feedback from review

jolelievre committed:
Use flex grow 0 to limit the cover size in flex container

matks reviewed (dismissed):
OK for me.
I'm not good enough to validate the CSS 🤷‍♂️

PierreRambaud reviewed (dismissed):
Unfortunately, you have to rebase your PR :D

jolelievre committed:
Fix new quantity field naming

jolelievre commented:
@matks @NeOMakinG @PierreRambaud PR has been rebased, it needs re-approval to go back to QA please :)

jolelievre committed:
Add margin top on header

khouloudbelguith reviewed (changes requested):
Hi @jolelievre,

I found some issues:
**Issue1**

1. Enable Experimental product page
2. Go to BO > Catalog > Products page
3. Edit any product V2
4. Click on the type of this product => A modal is opened
5. Click on the "Change product type" button without changing the product type
6. See an error 

![image](https://user-images.githubusercontent.com/16067358/151414510-9cc257cf-557c-4556-985b-e42a2468dedd.png)

**Second issue**
When using a mobile => the header is NOK, is this the scope of the PR? 
![image](https://user-images.githubusercontent.com/16067358/151414873-465c661e-18da-4b89-8940-d7eb78806cae.png)

**Third issue**
1. Enable Experimental product page
2. Go to BO > Catalog > Products page
3. Try to add a new product with V2
4. Choose a product with combination / a pack product or product with combinations and add a product
5. See error => the type of product is incorrect

https://user-images.githubusercontent.com/16067358/151415324-5958f4c0-2cb5-4a53-98c0-06599419fd7a.mp4

Thanks!

Hlavtox commented:
@khouloudbelguith I can do the responsivity after this is merged. 🙂

jolelievre committed:
Disable current type when switching product type

jolelievre committed:
Fix creation handler

jolelievre commented:
Thank you @khouloudbelguith I just fixed the two bugs:
- product creation selected type ignored
- type switching we disable the current type so that it can't be picked

About the mobile display, indeed it's not in the scope of this PR.

@Hlavtox if you are motivated to handle this that's great! Maybe in cooperation with @NeOMakinG Because the whole page is gonna need some responsive love at some point, not just the header (or footer ^^)

khouloudbelguith reviewed (changes requested):
Hi @jolelievre,

I have some issues:

**First issue**
1. Enable Experimental product page
2. Go to BO > Catalog > Products V2
3. Create a standard product or a Pack of products or a Virtual product
4. See an error in the header: 0 in green & in stock label 
![image](https://user-images.githubusercontent.com/16067358/151772784-282e80de-eb20-4952-9775-a57d021cf72e.png)
=> 
It should be `red background` & `out of stock label`
5. In the Stock Tab > Set Low stock level with 2 and save
6. See error in the header 
![image](https://user-images.githubusercontent.com/16067358/151773311-87e0bcb8-a355-4414-829c-bccf607cd021.png)
=> 
It should be `red background` & `out of stock label`

**Second issue**
1. Enable Experimental product page
2. Go to BO > Catalog > Products V2
3. Create a product with combinations
4. Generate combinations
5. The Stock is not displayed for this type 
![image](https://user-images.githubusercontent.com/16067358/151775697-1d4cf739-34c0-45b8-8072-8dc26d0b8850.png)


**Third issue**
It could be related to this issue: https://github.com/PrestaShop/PrestaShop/issues/27054

1. Enable Experimental product page
2. Go to BO > Catalog > Products V2
3. Create a product, in the Pricing Tab > Tax rule = `EU VAT For Virtual Products` & save
4. Refresh the page
5. See an error in the header > the tax rule is changing 

I attached a screen record
https://user-images.githubusercontent.com/16067358/151776764-a893d7af-345a-422e-9964-ad722a18da89.mp4

**Fourth issue**
1. Go BO > International > Localization > Import the Algeria pack
2. Change the employee profile to Arabic
2. Go to BO > Catalog > Products V2
3. Create a product V2 > Choose a type & add new product > See error 
![image](https://user-images.githubusercontent.com/16067358/151780157-91695359-25e2-4839-bfff-9af30805cf9d.png)
4. Edit a product V2 > See error  
![image](https://user-images.githubusercontent.com/16067358/151780195-757b41ac-2d86-4fba-a96a-fd51e71106c7.png)

**Fifth issue**
1. Go to BO >  International > Localization > Languages Tab
2. Create manually a language => in my case, I used this example
![image](https://user-images.githubusercontent.com/16067358/151780802-aefeaaba-25ea-4a54-9b12-165fff346068.png)
3. Change the employee language to this new language created
4. Go to BO > Catalog > Products V2
3. Create a product
4. See an error in the header => the currency symbol is not displayed
![image](https://user-images.githubusercontent.com/16067358/151781325-12e954dc-a4e5-40f8-a991-0139865d3e00.png)

**Sixth issue**

1. In fact when creating a product with a too-long data
2. In the header, the title of the product will be not well displayed as before => I think it could be a problem 
![image](https://user-images.githubusercontent.com/16067358/151782408-ca2eefde-7f22-4724-b639-32000d300d0f.png)
 before, it was 
![image](https://user-images.githubusercontent.com/16067358/151782733-1fea2f71-cb39-45c1-ab37-ba8c28e912b3.png)

**Seventh issue**
1. Go to BO > Shop Parameters > Product Settings > Disable `Enable stock management` 
![image](https://user-images.githubusercontent.com/16067358/151790300-2b344d2b-b1ba-4b11-aca3-20def10a4b7a.png)
2. Go to BO > Catalog > Products V2
3. Create a product V2 > Choose a type & add new product > See error 
![image](https://user-images.githubusercontent.com/16067358/151790367-0bdc09ee-9c57-4134-a543-bdb1464e58b5.png)
4. Edit a product V2 > See error  
![image](https://user-images.githubusercontent.com/16067358/151790478-f032e6a8-a82f-4b73-b83d-19ad2585ed57.png)


Thanks!

jolelievre committed:
Improve condition checking for stock state and use custom labels when possible

jolelievre committed:
Fix bug when displaying localized arabic values

jolelievre committed:
Make twig format price more resilient to custom Language with locale field

jolelievre committed:
Add missing length constraint on references

jolelievre committed:
Fix bug with loo long references

jolelievre committed:
Fix bug when stock management is disabled

jolelievre commented:
Thank @khouloudbelguith 

I fixed a few bugs from what you discovered:

### First issue
Nice catch my condition was wrong to detect the state, but most of all I didn't think about using the custom labels You're right it's better this way We need to confirm this with @MatShir and @TristanLDD though Besides there is no custom label for "low stock" so I always use the default wording

### Second issue
For combinations, I didn't display because each combination has its stock, and the Stock tab is not visible so I assumedthe preview worked the same If we decide to handle this use case différently it's another issue for me @MatShir what do you think?

### Third issue
We will indeed have the same bug as in #27054 because I use the same values and the bug has not been fixed However, what you showed in the video is différent the tax rate changes at each page refresh This one I was not able to reproduce The tax rate may be incorrect depending on the country but it's always the same for me What country did you choose in your Employee settings or in the default country?

### Fourth issue
I fixed as well as I could, the bug was because the value was converted in Arabic display like this for example `١٨٫٠٠٠٠٠٠` which is not usable for the CLDR So instead I used the float value but it's not exactly displayed in Arabic... I was inspired by the front behaviour where latin characters are used in the product previews as well So it makes sense that the preview follows the same principle. However, inputs and stock preview use Arabic characters but I don't know if it's the wanted behaviour? I'd like your opinion (and the one from @MatShir) on this, is this ok to mix both use of characters depending on the need? or should we use one type of character everywhere?

<img width="635" alt="Capture d’écran 2022-02-01 à 00 35 09" src="https://user-images.githubusercontent.com/13801017/151890614-a60b01bd-c824-4c2b-abbf-8313a4cdba80.png">

<img width="775" alt="Capture d’écran 2022-02-01 à 00 34 44" src="https://user-images.githubusercontent.com/13801017/151890598-fb617ae9-55a1-46a7-9559-3653f0851a7b.png">

### Fifth issue
Fixed based on the same fallback used on front.

### Sixth issue
I think the problem is not really about long names in this case, but rather long references. I updated the display a little so that all info is correctly displayed. Though I didn't go into details for all kind of screen sizes so when the width gets smaller and smaller some issues might happen. But this will have to be handled in the PR for responsiveness on this page. For now the quick fix seems enough for me.

MatShir commented:
> Hi @jolelievre,
> 
> I have some issues:
> 
> **First issue**
> 1. Enable Experimental product page
> 2. Go to BO > Catalog > Products V2
> 3. Create a standard product or a Pack of products or a Virtual product
> 4. See an error in the header: 0 in green & in stock label 
> ![image](https://user-images.githubusercontent.com/16067358/151772784-282e80de-eb20-4952-9775-a57d021cf72e.png)
> => 
> It should be `red background` & `out of stock label`
> 5. In the Stock Tab > Set Low stock level with 2 and save
> 6. See error in the header 
> ![image](https://user-images.githubusercontent.com/16067358/151773311-87e0bcb8-a355-4414-829c-bccf607cd021.png)
> => 
> It should be `red background` & `out of stock label`

The wording here is not meant to be adaptive to the label.

**Second issue**

> 1. Enable Experimental product page
> 2. Go to BO > Catalog > Products V2
> 3. Create a product with combinations
> 4. Generate combinations
> 5. The Stock is not displayed for this type 
> ![image](https://user-images.githubusercontent.com/16067358/151775697-1d4cf739-34c0-45b8-8072-8dc26d0b8850.png)
> 

For a product with combination, the stock displayed on the header will be the total amount of the combination. The low stock display won't be available here for this type of product.

**Fourth issue**

> 1. Go BO > International > Localization > Import the Algeria pack
> 2. Change the employee profile to Arabic
> 2. Go to BO > Catalog > Products V2
> 3. Create a product V2 > Choose a type & add new product > See error 
> ![image](https://user-images.githubusercontent.com/16067358/151780157-91695359-25e2-4839-bfff-9af30805cf9d.png)
> 4. Edit a product V2 > See error  
> ![image](https://user-images.githubusercontent.com/16067358/151780195-757b41ac-2d86-4fba-a96a-fd51e71106c7.png)
> 

All the RTL and Arabic language compliance is not yet supported, we will create a dedicated issue to handle it :)

jolelievre committed:
Out of stock labels are always fixed

jolelievre committed:
Display global stock for product with combinations

jolelievre committed:
Improve tax rate selection for summary

jolelievre commented:
Hi @khouloudbelguith and @MatShir 

based on our discussion with Mateus and his decisions I made the following modifications:
- the product with combinations: the stock is displayed it is the sum of all combinations' quantity (still not updated dynamically with JS only on refresh) We don't handle low stock state for combinations
- we always use the same out of stock labels regardless of their customized value
- I wasn't able to reproduce your bug with tax rate that changes at each refresh, but I made a small modif in this component so maybe it will fix the bug on your side @khouloudbelguith

jolelievre commented:
Hi @matks @matthieu-rolland and @NeOMakinG 

I need your re-approval on this one after rebase to resolve conflicts, nothing changed

khouloudbelguith reviewed (changes requested):
Hi @jolelievre,

I have some issues:
**First issue**

1. Enable Experimental product page
2. Go to BO > Catalog > Products V2
3. Create a new product
4. See error => the `product-field-label` "tax included" is not displayed & the` tax rule = 0` is displayed

![image](https://user-images.githubusercontent.com/16067358/152782697-8364429c-d5c3-45e5-a50d-c7f8696b598e.png)

**Before**, we have this
![image](https://user-images.githubusercontent.com/16067358/152783173-f4babcb6-6dde-4490-a55f-b31ad9305e79.png)

**Second issue**
1. Enable Experimental product page
2. Go to BO > Catalog > Products V2
3. Edit a standard simple product with stock different from 0 
![image](https://user-images.githubusercontent.com/16067358/152783530-8a969a2d-0ef9-4169-9962-286365d3d80d.png)
4. Change product type to "Product with combinations"
5. In the Combinations Tab > Generate some combinations 
6. Refresh the page
7. See error => the stock in the header is not updated 
![image](https://user-images.githubusercontent.com/16067358/152783813-324c7ff3-d5f0-4dee-a6ac-8d7e4e289207.png)

Is it normal behavior because of this?
`the product with combinations: the stock is displayed it is the sum of all combinations' quantity (still not updated dynamically with JS only on refresh) We don't handle low stock state for combinations`

I checked FO => OK

>Third issue
We will indeed have the same bug as in https://github.com/PrestaShop/PrestaShop/issues/27054 because I use the same values and the bug has not been fixed However, what you showed in the video is différent the tax rate changes at each page refresh This one I was not able to reproduce The tax rate may be incorrect depending on the country but it's always the same for me What country did you choose in your Employee settings or in the default country?


About this issue, I think because if I edit this tax rule => I have different taxes for different countries
![image](https://user-images.githubusercontent.com/16067358/152784410-ff634d46-631c-4c66-9fd0-40bd367356f7.png)

Thanks!

Hlavtox commented:
@khouloudbelguith Don't forget to check how it behaves with VAT disabled. A lot of new merchants are not VAT payers.

khouloudbelguith commented:
> @khouloudbelguith Don't forget to check how it behaves with VAT disabled. A lot of new merchants are not VAT payers.

@Hlavtox, thanks a lot :slightly_smiling_face:

jolelievre committed:
Fix wording when tax rate is zero

jolelievre committed:
Add behat tests for global stock on product with combinations

jolelievre committed:
Reset product stock when product type is switched to combinations

jolelievre committed:
Display new warning for stock when switching to product with combinations

jolelievre committed:
Check that stock is not zero to avoid constraint exception, and unnecessary updates

jolelievre commented:
Hi @khouloudbelguith 

I fixed the first two issues, for stock it s now reset when the product is switched to Combinations type This was discussed with @MatShir the spec needs to be updated accordingly. I added a warning in the modal as well with a temporary text:
<img width="878" alt="Capture d’écran 2022-02-16 à 18 00 53" src="https://user-images.githubusercontent.com/13801017/154318989-0ba739f3-0214-4799-be2c-c642650e17f1.png">

The wording will need to be approved or improved by @Julievrz 

However, regarding the third issue @khouloudbelguith I am still unable to reproduce it. I also have different tax rates based on the country for this tax rule but still, I always get 24%.

Maybe it is related to the shop default country, or the country of the employee. What are the settings for those two configurations? (I use English as my employee's language but France as the shop's default country)

khouloudbelguith commented:
Hi @jolelievre,

>However, regarding the third issue @khouloudbelguith I am still unable to reproduce it. I also have different tax rates based on the country for this tax rule but still, I always get 24%.
Maybe it is related to the shop default country, or the country of the employee. What are the settings for those two configurations? (I use English as my employee's language but France as the shop's default country)


Yes, In my case, I installed this PR with English language & France country.

Thanks!

jolelievre commented:
@khouloudbelguith ok it's weird same for me...

If it's the only issue blocking at the end maybe we can open a dedicated issue, to see if someone else reproduces it Because right now I can't do much sadly 🤷‍♂️

jolelievre committed:
Improve pack scenarios reference names, and fix restoreProductTables which forgot to reset pack table

matks commented:
@jolelievre what do we do for this PR?

jolelievre committed:
Handle reset stock for multishop

jolelievre committed:
Update behat tests for reset on multishop

jolelievre commented:
@matks I had to rebase it and fix the stock reset for multishop case in behat tests It's reviewable again now! (@PierreRambaud if you can re-approve it as well please?)

@khouloudbelguith I tried to reproduce the bug about switching tax rate again without success I don't think it should block this PR instead I suggest adding this use case to this issue https://github.com/PrestaShop/PrestaShop/issues/27054 since it's probably related Is it ok for you?

Progi1984 commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -109,6 +109,112 @@ $product-page-padding-bottom: 80px !default;
+
+    .product-header-cover {
+      flex-grow: 0;
+      min-width: $product-cover-size;
+      max-width: $product-cover-size;
```
If you define min & max at the same size, why not define directly the `width` ?

jolelievre replied:
Because the element is in a flex container so its width adapts automatically to the other elements Basically the `width` attribute is not really taken into account So the way I found to limit the size of this element was with min and max

Progi1984 replied:
Ok. May be @NeOMakinG has a best solution ?

Hlavtox replied:
Don't lose time with CSS, just make it so it works code wise, we will do the styling and HTML with Valentin if needed. 👍

Progi1984 commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -1058,11 +1138,61 @@ $product-page-padding-bottom: 80px !default;
   }
 }
 
+// We use this class to scope the modifications that only target product page v2
```
```suggestion
// We use this class to scope modifications that only target product page v2
```

jolelievre replied:
I think both are correct, or at least understandable The PR already has one approval so if it's the only feedback I'm not sure it's worth blocking the PR for this 😅

Progi1984 commented on src/PrestaShopBundle/Twig/Extension/LocalizationExtension.php:
```diff
@@ -43,24 +46,72 @@ class LocalizationExtension extends AbstractExtension
+    public function __construct(
+        string $contextDateFormatFull,
+        string $contextDateFormatLight,
+        Repository $localeRepository,
+        $contextLanguage,
+        ?Currency $contextCurrency
```
Please define BC Breaks

Progi1984 commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Blocks/header-details.html.twig:
```diff
@@ -0,0 +1,106 @@
+
+        {# If tax rate is defined display it in the label #}
+        {% if taxRate %}
+          {{ 'tax incl. (tax rule: %taxRate%%)'|trans({'%taxRate%': taxRate|round(2)}, 'Admin.Catalog.Feature') }}
+        {% else %}
+          {{ 'tax incl.'|trans({'%taxRate%': taxRate|round(2)}, 'Admin.Catalog.Feature') }}
```
Shouldn't we use the precision rather than `2` ?

jolelievre replied:
Nope, it's not a price which is displayed here It's purely informational to remind the value of the tax rate so 2 decimals are more than enough IMHO

Progi1984 commented on src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php:
```diff
@@ -98,30 +73,18 @@ class ProductFormDataProvider implements FormDataProviderInterface
      * @param int $defaultShopId
      * @param int|null $contextShopId
      */
     public function __construct(
```
Could you define the BC Break ?

Progi1984 commented on src/Adapter/Product/Update/ProductTypeUpdater.php:
```diff
@@ -60,22 +62,30 @@ class ProductTypeUpdater
         CombinationDeleter $combinationDeleter,
-        VirtualProductUpdater $virtualProductUpdater
+        VirtualProductUpdater $virtualProductUpdater,
+        ProductStockUpdater $productStockUpdater
```
Could you define the BC Break ?

Progi1984 commented on src/Adapter/Product/Stock/Update/ProductStockUpdater.php:
```diff
@@ -80,17 +86,20 @@ class ProductStockUpdater
         StockManager $stockManager,
         ProductMultiShopRepository $productRepository,
         StockAvailableMultiShopRepository $stockAvailableRepository,
+        MovementReasonRepository $movementReasonRepository,
```
Could you define the BC Break ?

Progi1984 reviewed (commented):
Some feedbacks

atomiix commented on src/Adapter/Product/Stock/Update/ProductStockUpdater.php:
```diff
@@ -118,6 +128,57 @@ public function update(ProductId $productId, ProductStockProperties $properties,
+     */
+    public function resetStock(ProductId $productId, ShopConstraint $shopConstraint): void
+    {
+        if ($shopConstraint->forAllShops()) {
+            $shops = $this->productRepository->getAssociatedShopIds($productId);
+        } else {
+            $shops = [$shopConstraint->getShopId()];
+        }
```
What about groups? 🤔

matthieu-rolland replied:
I'm not sure either... 

For now this method is always given a "all stop" context, and the product page cannot be accessed in group shop, maybe it's the reason ?

@jolelievre if you could enlighten us ? 🤔

jolelievre replied:
Product entity cannot be edited in the shop group context which is why only single and All shops context are handled
In other parts of the code I added a check though that throws an exception when a group constraint is used, I'll add this safety here as well!

jolelievre replied:
Even if technically for now the method is only called with an All shops constraint, I built in a way that can be potentially used for other use cases I didn't want to explicitly bind it to the All shops context

So it's a lot better to add this extra check indeed!

matthieu-rolland reviewed (commented):
LGTM overall, there's just a clarification is needed about shop context and stock reset

jolelievre committed:
Throw exception in case shop group constraint is used to reset stock

jolelievre commented:
@Progi1984 @atomiix @matthieu-rolland I just handled your feedback it should be fine now

If you have other questions don't hesitate to direct message me, I'd like this one to be approved as fast as possible since it brings a lot of value to the page (and many conflicts to be expected on other PRs so it's better to anticipate them)

khouloudbelguith reviewed (dismissed):
Hi @jolelievre,

Could you please check the Conflicting files?

Thanks!

jolelievre commented:
Thanks @khouloudbelguith 

I just rebased and fixed the conflicts it should be good now If @PierreRambaud @Progi1984 and/or @NeOMakinG can reapprove it it's ready for QA

atomiix commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Blocks/header-details.html.twig:
```diff
@@ -0,0 +1,106 @@
+
+<div class="product-header-details">
+{% block header_summary %}
+  {% set productData = productForm.vars.data %}
+  <div class="product-header-summary">
+    <div class="product-field-preview">
+      <div class="product-field-value">
+        {# It is important to use vars.data and not vars.value because value contains the localized value which may
+         # not be written with numeric characters anymore depending on the language but data is the initial float data.
+         #}
```
Just a suggestion to make the comment closer to the line related
```suggestion
  {# It is important to use vars.data and not vars.value because value contains the localized value which may
   # not be written with numeric characters anymore depending on the language but data is the initial float data.
   #}
  {% set productData = productForm.vars.data %}
  <div class="product-header-summary">
    <div class="product-field-preview">
      <div class="product-field-value">
```

jolelievre committed:
Better comment

jolelievre commented:
Thanks @atomiix I moved the comment

If you and someone else could approve this PR it would be great, it's been waiting to go back to its final QA for more than a month now 😅

jolelievre commented:
@matks or @NeOMakinG you already approved this one in the past, if you can do it again it would be awesome ^^

khouloudbelguith reviewed (dismissed):
Hi @jolelievre,

I checked this PR with @HanaRebaiQA, we found those issues:

Specs issues:

**First issue**
Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type
>Change product type - Button - Save the new product type, **set the product as offline**, and refresh the page. If the user changes the product type, only the common data between the product type will be saved.

1. Go to BO > Products V2
2. Edit any enabled product
3. Click on the type of the product to change it
4. Select any type and click on "Change product type"
5. See the alert info: Changing the product type will immediately save the product and refresh the page. => OK
6. See error: the product is not disabled as described in the spec


https://user-images.githubusercontent.com/16067358/158423102-a77c8b25-26fc-4253-a487-9f09037368d8.mp4

**Second issue**
Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type

| Changing the Product type form| Impact | warning-alert in the Confirmation modal |
| --- | --- | --- |
| Standard product | No impact, the data is common between the standard product and the others products. | |

1. Go to BO > Products V2
2. Edit a standard product with quantity <> 0 
3. Change the type of product to Product with combinations
4. In the Confirmation Modal => there is a warning 
![image](https://user-images.githubusercontent.com/16067358/158424268-f9f3e2f6-70a5-4fb1-a98c-13b15b039477.png)

Ping @MatShir, we think we need to update the spec.

**Third issue**
Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type

| Changing the Product type form| Impact | warning-alert in the Confirmation modal |
| --- | --- | --- |
| Standard product | No impact, the data is common between the standard product and the others products. | |
| Product with combination | Deletes the combination and reset the stock to 0. | This will delete all combinations. <br>This will reset the stock of this product. |

1. Go to BO > Catalog > Products V2
2. Edit a product with combinations (quantity <> 0)
3. Change the type of product to another type
4. In the Confirmation Modal => there is only one warning `This will delete all combinations.`

=> the warning `This will reset the stock of this product.` is missing
![image](https://user-images.githubusercontent.com/16067358/158425424-58b5fc78-4f58-4744-b0a9-a92fa6e7cbb6.png)

-----------------------------------------------------------------
Other issues

**Fourth issue**
1. Go to BO > International > Taxes page
2. Disable Tax 
![image](https://user-images.githubusercontent.com/16067358/158426879-040fd647-ba34-424d-b98d-0097d4e3faab.png)
3. Add a new product V2 > In the pricing tab > Set a Retail price and a tax rule = 20% for example
4. Save
5. See error: in the header: the `tax incl.` is incorrect, it should be the same `tax excl.` 
![image](https://user-images.githubusercontent.com/16067358/158435146-8b770564-abad-4934-a0f2-466170224056.png)

PS: check in the FO 
![image](https://user-images.githubusercontent.com/16067358/158430029-e8aa3205-d1bc-48fc-a802-040957d61e0f.png)

Ping @MatShir what do you think?

**Fifth issue**
1. Go to BO > International > Taxes > Tax rules page
2. Create a new tax rule with the behavior: Combine & One After Another
```
Behavior. Some customers might have an address that matches more than one of your tax rules. In that case, you can choose how this tax rule should behave:
This Tax Only. Will apply only this tax, not any of the other matching taxes.
Combine. Combine taxes. For instance: 100€ + (10% + 5% => 15%) => 115€.
One After Another. Apply taxes one after another. For instance: 100€ + 10% => 110€ + 5% => 115.5€.
```
![image](https://user-images.githubusercontent.com/16067358/158430736-254525bc-d1d4-4ba5-a7a9-d286a524ae4a.png)
3. Create a product V2 with this tax rule and save
![image](https://user-images.githubusercontent.com/16067358/158431403-c6129e39-f07d-4a60-8642-b9d887005017.png)
4. See an error in the header => 
![image](https://user-images.githubusercontent.com/16067358/158431864-72a81d1f-c8df-4e6b-a063-3c17f859fa44.png)

PS: in the FO => OK 
![image](https://user-images.githubusercontent.com/16067358/158432121-6773687f-912b-45e0-8582-965ee4a94e5c.png)

**Sixth issue**
1. Go to BO > Shop Parameters > Product Settings > Disable `Enable stock management` 
![image](https://user-images.githubusercontent.com/16067358/151790300-2b344d2b-b1ba-4b11-aca3-20def10a4b7a.png)
2. Go to BO > Catalog > Products V2
3. Edit a product
4. In the header, the stock is displayed, but in the Stock tab, there is no stock displayed 
![image](https://user-images.githubusercontent.com/16067358/158433607-bf7656ea-e43d-4e87-9b2f-d0ee9dc13b9a.png)

We think it is a bug, ping @MatShir what do you think?

Thanks!

jolelievre committed:
Do not display tax or stock in header when features are disabled

jolelievre committed:
Handle price tax included and excluded are equals when Tax feature is disabled

jolelievre committed:
Display warning and correctly reset product stock when switch FROM type with combinations

jolelievre committed:
Fix shop group behat context to use references everywhere

MatShir commented:
> Hi @jolelievre,
> 
> I checked this PR with @HanaRebaiQA, we found those issues:
> 
> Specs issues:
> 
> **First issue** Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type
> 
> > Change product type - Button - Save the new product type, **set the product as offline**, and refresh the page. If the user changes the product type, only the common data between the product type will be saved.
> 
> 1. Go to BO > Products V2
> 2. Edit any enabled product
> 3. Click on the type of the product to change it
> 4. Select any type and click on "Change product type"
> 5. See the alert info: Changing the product type will immediately save the product and refresh the page. => OK
> 6. See error: the product is not disabled as described in the spec
> 
>  Products.experimental.PrestaShop.mp4 

This is the expected behavior. I thought the page went offline when you change the product in the legacy. And we agreed with Jo to keep this behavior as it is in the PR. 
I did a PR to update the specs.
→ https://github.com/PrestaShop/prestashop-specs/pull/316

> **Second issue** Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type
> 
> Changing the Product type form	Impact	warning-alert in the Confirmation modal
> Standard product	No impact, the data is common between the standard product and the others products.	
> 1. Go to BO > Products V2
> 2. Edit a standard product with quantity <> 0
> 3. Change the type of product to Product with combinations
> 4. In the Confirmation Modal => there is a warning
>    ![image](https://user-images.githubusercontent.com/16067358/158424268-f9f3e2f6-70a5-4fb1-a98c-13b15b039477.png)
> 
> Ping @MatShir, we think we need to update the spec.

Specs updated :) → → https://github.com/PrestaShop/prestashop-specs/pull/316
Thanks ;)

> **Third issue** Spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#change-of-product-type
> 
> Changing the Product type form	Impact	warning-alert in the Confirmation modal
> Standard product	No impact, the data is common between the standard product and the others products.	
> Product with combination	Deletes the combination and reset the stock to 0.	This will delete all combinations.
> This will reset the stock of this product.
> 1. Go to BO > Catalog > Products V2
> 2. Edit a product with combinations (quantity <> 0)
> 3. Change the type of product to another type
> 4. In the Confirmation Modal => there is only one warning `This will delete all combinations.`
> 
> => the warning `This will reset the stock of this product.` is missing ![image](https://user-images.githubusercontent.com/16067358/158425424-58b5fc78-4f58-4744-b0a9-a92fa6e7cbb6.png)

@jolelievre will fix it if it is not already fixes and the specs are updated as well :) → https://github.com/PrestaShop/prestashop-specs/pull/316

> Other issues
> 
> **Fourth issue**
> 
> 1. Go to BO > International > Taxes page
> 2. Disable Tax
>    ![image](https://user-images.githubusercontent.com/16067358/158426879-040fd647-ba34-424d-b98d-0097d4e3faab.png)
> 3. Add a new product V2 > In the pricing tab > Set a Retail price and a tax rule = 20% for example
> 4. Save
> 5. See error: in the header: the `tax incl.` is incorrect, it should be the same `tax excl.`
>    ![image](https://user-images.githubusercontent.com/16067358/158435146-8b770564-abad-4934-a0f2-466170224056.png)
> 
> PS: check in the FO ![image](https://user-images.githubusercontent.com/16067358/158430029-e8aa3205-d1bc-48fc-a802-040957d61e0f.png)
> 

@jolelievre will fix it if it is not already fixes and the specs are updated as well :) → https://github.com/PrestaShop/prestashop-specs/pull/316> 

> **Fifth issue**
> 
> 1. Go to BO > International > Taxes > Tax rules page
> 2. Create a new tax rule with the behavior: Combine & One After Another
> 
> ```
> Behavior. Some customers might have an address that matches more than one of your tax rules. In that case, you can choose how this tax rule should behave:
> This Tax Only. Will apply only this tax, not any of the other matching taxes.
> Combine. Combine taxes. For instance: 100€ + (10% + 5% => 15%) => 115€.
> One After Another. Apply taxes one after another. For instance: 100€ + 10% => 110€ + 5% => 115.5€.
> ```
> 
> ![image](https://user-images.githubusercontent.com/16067358/158430736-254525bc-d1d4-4ba5-a7a9-d286a524ae4a.png) 3. Create a product V2 with this tax rule and save ![image](https://user-images.githubusercontent.com/16067358/158431403-c6129e39-f07d-4a60-8642-b9d887005017.png) 4. See an error in the header => ![image](https://user-images.githubusercontent.com/16067358/158431864-72a81d1f-c8df-4e6b-a063-3c17f859fa44.png)

This edge case will be handled in this issue https://github.com/PrestaShop/PrestaShop/issues/27054.

> 
> PS: in the FO => OK ![image](https://user-images.githubusercontent.com/16067358/158432121-6773687f-912b-45e0-8582-965ee4a94e5c.png)
> 
> **Sixth issue**
> 
> 1. Go to BO > Shop Parameters > Product Settings > Disable `Enable stock management`
>    ![image](https://user-images.githubusercontent.com/16067358/151790300-2b344d2b-b1ba-4b11-aca3-20def10a4b7a.png)
> 2. Go to BO > Catalog > Products V2
> 3. Edit a product
> 4. In the header, the stock is displayed, but in the Stock tab, there is no stock displayed
>    ![image](https://user-images.githubusercontent.com/16067358/158433607-bf7656ea-e43d-4e87-9b2f-d0ee9dc13b9a.png)
> 
> We think it is a bug, ping @MatShir what do you think?
> 
> Thanks!

All the quantity field should be hidden. For this PR, the missing part is will be the "Behavior out of the stock".
I'm going to create a dedicated issue for it.

![Capture d’écran 2022-03-16 à 16 34 48](https://user-images.githubusercontent.com/43613217/158628254-f970677e-931b-4a31-8445-7f824cf816f7.png)

HanaRebaiQA reviewed (dismissed):
Hello @jolelievre 

Thanks for the fixed issues.
@khouloudbelguith and me are checked again this PR and here is our feedback :

**First issue :** 
The stock is not updated when editing a product with combinations using a new shop

1. Go to BO > Catalog > Products V2 
2. Create a product `with Combinations`, generate combinations for this product and with `quantity <> 0`
3. Enable multistore and create a new shop
4. Edit the created product with the new shop 
5. Edit quantities for every combinations and Save
6. See error >> The total of stock is NOK 
![image](https://user-images.githubusercontent.com/92912932/159305687-e067ac2f-667e-4b8f-9ee6-aa9ea94f2901.png)

https://user-images.githubusercontent.com/92912932/159305368-9340da9e-df9a-4d8a-b974-e5b9f32d2d1b.mp4


**Second issue :** 
The combinations are not erased (still displayed ) when changing the type of product from `Combinations > Any other type > Changing it again to Combinations` when using a new shop already created


1. Go to BO > Catalog > Products V2 
2. Create a product `with Combinations` and generate combinations for this product 
3. Enable multistore and create a new shop
4. Select the new shop created and edit the last product created with V2
5. Change the type of product to any other type (eg.Standard)
6. Then, change the type again to product `with Combinations`
7. See error >> The generated Combinations of the fist product are not erased (still displayed)

https://user-images.githubusercontent.com/92912932/159316035-95bba019-6670-47a7-9fe7-1d3d17be934d.mp4

**Thrid issue :** 
The tax is updated with every refresh of the page when using theses configuration for the tax rule : 
1. Use different taxes for different Countries 
![image](https://user-images.githubusercontent.com/92912932/159333091-00a61ca6-b9f4-4a74-a2bd-08cb8de06fef.png)
2. Use a One by One Tax 
![image](https://user-images.githubusercontent.com/92912932/159333534-8f23a6dd-1c60-4193-bf04-c0bda626f003.png)

3. Use a tax with Combine 
![image](https://user-images.githubusercontent.com/92912932/159333381-22994e5f-b70d-425e-8a11-a65dc6f9be56.png)

4. Use different taxes for different States as described [here](https://github.com/PrestaShop/prestashop-specs/pull/316/files?diff=split&w=0#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R47)

ping @jolelievre This issue is already reproduced https://github.com/PrestaShop/PrestaShop/pull/27233#issuecomment-1055353950 What do you think ? it should be fixed [here](https://github.com/PrestaShop/PrestaShop/issues/27054)?


https://user-images.githubusercontent.com/92912932/159329718-3a369bba-f618-4337-b62f-6c289e612e6a.mp4


**Fourth issue :** 

The stock is displayed with a green color when the quantity is equal to the low stock level 
But, in specs, the green color must be displayed when the quantity **is strictly superior** to the low stock level
see specifications : 
https://github.com/PrestaShop/prestashop-specs/pull/316/files#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R56-R58
![image](https://user-images.githubusercontent.com/92912932/159323885-ba5311e5-7460-455a-8e69-964d059c3e6e.png)

1. Go to BO > Catalog > Products V2 
2. Create a standard product equal to low stock level 
3. See error >> The stock is displayed in green color

![image](https://user-images.githubusercontent.com/92912932/159328389-1e4d0aa2-42f7-413e-8955-6d101681cd68.png)

https://user-images.githubusercontent.com/92912932/159326314-658e76c7-89ce-47a7-9ce0-1193e38ca036.mp4

Ping @MatShir what do you think? 

**Fifth issue :** 
The warning message `This will reset the stock of this product` is displayed in case the stock is **disabled** 

1. Go to BO > CONFIGURE > Shop parameters > Product settings
2. Disable the `Enable stock management` parameter
3. Go to BO > Catalog > Products V2
4. Edit a product 
5. Change from any product type to product with combinaitions OR from product with combinations to any other type
6. See error >> The stock warning is displayed but it should'nt
ping @MatShir What do you think ?

https://user-images.githubusercontent.com/92912932/159329471-bd54c477-32f0-4537-95d9-9ea9d9765f84.mp4


![image](https://user-images.githubusercontent.com/92912932/159329436-d31ee76b-0374-44ee-954a-8483c6917d2e.png)

**Specs issues**

**Sixth issue :** 
https://github.com/PrestaShop/prestashop-specs/pull/316/files#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R96
| Changing the Product type to | Impact | warning-alert in the Confirmation modal |
| --- | --- | --- |
| Product with combination | Deletes the combination and reset the stock to 0. | This will reset the stock of this product. |

The Impact column is NOK, we are changing the type of the product **to** product with combinations, so we don't have yet the combinations to delete them 
ping @MatShir  what do you think?

**Seventh issue :** 
https://github.com/PrestaShop/prestashop-specs/pull/316/files#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R44-R45
>IF the tax rule has not a tax for the shop's country
THEN it displays on the Tax label 0% next to the country ISO code.
![image](https://user-images.githubusercontent.com/92912932/159332057-5f07699d-1e75-43f1-a806-111be6d82b8c.png)

ping @MatShir What do you think ?

Thanks!

jolelievre commented:
Hi @HanaRebaiQA and @khouloudbelguith 

thanks for your feedback. Here are my answers for the various issues:

#### First issue and the second issue
The multishop feature development is on pause for now so we can focus on finishing the development of the page in single shop mode. So I think we should create a dedicated issue for those two bugs.

#### Third issue and seventh issue
The problems related to taxes will be handled in this issue indeed https://github.com/PrestaShop/PrestaShop/issues/27054 There has already been an issue added into this one (related to combined taxes) But I think you can add this issue as it seems to be slightly different. This is also the issue where we will add the country code next to the tax.

#### Fourth and the fifth issue
To be fixed in this PR, I'll look into them

#### Sixth issue
To see with @MatShir but it seems like a mistake indeed.

jolelievre committed:
Low stock when stock is equal to low stock value

jolelievre committed:
Only display stock warning on type switching when the stock management is enabled

jolelievre committed:
Fix multishop SEO behat tests

HanaRebaiQA reviewed (dismissed):
Hello @jolelievre 

I checked again this PR : 
-The issues mentionned above are fixed : OK


I just found new issues related to **specs** : 

**First issue :** 
In this section, this condition is missing 
`If the quantity <= lowStockLevel, the color background is Orange `
![image](https://user-images.githubusercontent.com/92912932/160447414-1fb72d5b-8dc5-461e-a39d-b2b82616d11c.png)

@MatShir  Could you please check and update the spec in this part https://github.com/PrestaShop/prestashop-specs/pull/316/files#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R56-R64 ?

**Second issue :** 
When the `Enable stock management` parameter is disabled, there is Only one warning when change the product from Combinations to other product type
`This will delete all combinations`
![image](https://user-images.githubusercontent.com/92912932/160447716-dc17c6d2-fc3a-4849-ad52-ca6de5e58e17.png)
@MatShir Could you please update the spec here https://github.com/PrestaShop/prestashop-specs/pull/316/files#diff-69aed766389514c34c80d6bac9113d027cc8640ae5e27ab4f7e6c478c0123a33R89-R97 ?

So, this PR will be QA :heavy_check_mark: when the specs are fine.

Thank you!

jolelievre committed:
Simplify product form data provider default data

jolelievre committed:
Fix conflicts from product form data provider

MatShir commented:
[Updated](https://github.com/PrestaShop/prestashop-specs/pull/316/commits/562b47b97653475041c59da9d11b2df965211606). Let's merge it ! :)

HanaRebaiQA reviewed (approved):
Hello,

Thank you @MatShir for the fix
@jolelievre It's OK for me after a new check on the PR
So, QA :heavy_check_mark: 

GO for the merge of the PR 
Thanks!

Progi1984 commented:
Thanks @jolelievre & @HanaRebaiQA
